core/src/main/java/cz/fi/muni/pv217/employee/management/OrderResource.java

```diff
@@ -0,0 +1,95 @@
+package cz.fi.muni.pv217.employee.management;
+
+import cz.fi.muni.pv217.employee.management.entity.Order;
+import cz.fi.muni.pv217.employee.management.service.OrderService;
+import io.quarkus.panache.common.Parameters;
+
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.JsonObject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+@Path("/api/order")
+@ApplicationScoped
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class OrderResource {
+
+    @Inject
+    OrderService orderService;
+    
+    @POST
+    @Path("/create")
+    @RolesAllowed("admin")
+    public Order createOrder(Order order) {
+        return orderService.createOrder(order);
+    }
+
+    @PUT
+    @Path("/{id}/update")
+    @RolesAllowed("admin")
+    public Order updateOrder(@org.jboss.resteasy.annotations.jaxrs.PathParam long id, JsonObject jsonObject) {
+        try {
+            return orderService.updateOrder(id, jsonObject);
+        } catch (IllegalArgumentException iae) {
+            throw new ClientErrorException(iae.getMessage(), Response
+                    .status(Response.Status.PRECONDITION_FAILED)
```

Comment:
Just to clarify, which status code is this?

Reply:
412